components/gitpod-db/src/typeorm/migration/1717407555612-WorkspaceDeletionEligibilityTime.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists, indexExists } from "./helper/helper";
+
+const TABLE_NAME = "d_b_workspace";
+const COLUMN_NAME = "deletionEligibilityTime";
+const INDEX_NAME = "ind_deletionEligibilityTime";
+
+export class WorkspaceDeletionEligibilityTime1717407555612 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, TABLE_NAME, COLUMN_NAME))) {
+            await queryRunner.query(
+                `ALTER TABLE \`${TABLE_NAME}\` ADD COLUMN \`${COLUMN_NAME}\` varchar(255) NOT NULL DEFAULT '', ALGORITHM=INPLACE, LOCK=NONE`,
```

Comment:
30 chars is enough for the timestamp (I know we cargo-culted 255 in a lot of places already, but still: this keeps indexes small):
```suggestion
                `ALTER TABLE \`${TABLE_NAME}\` ADD COLUMN \`${COLUMN_NAME}\` varchar(30) NOT NULL DEFAULT '', ALGORITHM=INPLACE, LOCK=NONE`,
```